lib/spack/spack/ci.py

```diff
@@ -505,10 +494,9 @@ def generate_gitlab_ci_yaml(env, print_summary, output_file,
         if not custom_spack_ref:
             custom_spack_ref = 'master'
         before_script = [
-            ('git clone "{0}" --branch "{1}" --depth 1 '
```

Comment:
@tgamblin @opadron This way of cloning made it impossible to provide a SHA rather than a named branch or tag.  Yet I feel we need the ability to specify a SHA so that we can avoid situations like what happened recently with a `develop` pipeline, where a job later in the pipeline got something different because `develop` changed.  I removed the `--depth` option because I want to always be sure we can reach back to the SHA we need on any branch.